### PR TITLE
Adding Reducers and Actions

### PR DESCRIPTION
**Pull request by taldr27**
### BookStore: Adding Reducers and Actions
In this milestone of the BookStore project I:
- Refactored directories to follow the duck pattern.
- Added `books` and `categories` files.
- Added reducers and actions to each file.
- Added a file `configureStore `to configure both reducers.

### CODE REVIEW

File: src/redux/books/Books.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      return [
+        ...state,
+        {
+          payload: action.payload,
+        },
+      ];
+    default:
+      return state;
+  }
+}
```
Comment (Nemwel-Boniface) [inline src/redux/books/Books.js]: - [x] Great job so far! I have realized that you are using both the ```ADD)BOOK``` and ```REMOVE_BOOK``` actions types however, I have rechecked and realized that you had not defined them anywhere. Kindly ensure that you define the action types and do not forget to use the [duck pattern](https://github.com/erikras/ducks-modular-redux) as well.

File: src/redux/books/Books.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          payload: action.payload,
+        },
+      ];
+    case 'REMOVE_BOOK':
+      return [
+        ...state,
+        {
+          payload: action.payload,
+        },
+      ];
```
Comment (Nemwel-Boniface) [inline src/redux/books/Books.js]: - [x] Kindly implement the deleting a book action making use of the JavaScript filter method.

File: src/redux/categories/Categories.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    type: 'CHECK_STATUS',
+    payload: 'Test',
+  }
+);
+
+export default function categoriesReducer(state = [], action) {
+  if (action.type === 'CHECK_STATUS') {
+    return 'Under Construction';
+  }
+  return state;
```
Comment (Nemwel-Boniface) [inline src/redux/categories/Categories.js]: - [x] I have realized that you are using the ```CHECK_STATUS``` action type here but it is not being defined anywhere. Kindly add the check status action type and also use the duck's pattern on it as mentioned earlier.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @taldr27      :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
### Hello @taldr27 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

**Pull request merged**